### PR TITLE
Remove unused methods

### PR DESCRIPTION
**Pull request by jnm2**
(None of these changes can be detected by users of the NUnit library.)

### CODE REVIEW

File: src/NUnitFramework/framework/Internal/TestExecutionContext.cs [hunk 1]
```diff
@@ -568,7 +568,7 @@ public class AdhocContext : TestExecutionContext
             public AdhocContext()
             {
                 var type = GetType();
-                var method = type.GetMethod("AdhocTestMethod", BindingFlags.NonPublic | BindingFlags.Instance);
+                var method = type.GetMethod(nameof(AdhocTestMethod), BindingFlags.NonPublic | BindingFlags.Instance);
```
Comment (jnm2) [inline src/NUnitFramework/framework/Internal/TestExecutionContext.cs]: This lets ReSharper know that `AdhocTestMethod` is not unused.

**Comment by jnm2** [comment]
Note to self: https://github.com/nunit/nunit/compare/master...jnm2:refactor_provider_cache?expand=1 depends on this

**Comment by jnm2** [comment]
@ChrisMaddock Do you think we should remove the `awaiting` label at the same time as applying the `closed` label?

**Comment by ChrisMaddock** [comment]
Can do...I was just being lazy. 🤷‍♂️😊

**Pull request merged**